include/boost/gil/image_processing/numeric.hpp

```diff
@@ -8,10 +8,14 @@
 #ifndef BOOST_GIL_IMAGE_PROCESSING_NUMERIC_HPP
 #define BOOST_GIL_IMAGE_PROCESSING_NUMERIC_HPP
 
-#include <boost/gil/detail/math.hpp>
-#include <cmath>
+#include <boost/gil/extension/numeric/kernel.hpp>
+#include <boost/gil/extension/numeric/convolve.hpp>
 #include <boost/gil/image_view.hpp>
 #include <boost/gil/typedefs.hpp>
+#include <boost/gil/detail/math.hpp>
+// fixes ambigious call to std::abs, https://stackoverflow.com/a/30084734/4593721
+#include <cstdlib>
```

Comment:
Not a request for change, just a remark: it would be better and clearer for future readers to do

```cpp
#if (defined(BOOST_CLANG) || defined(BOOST_GCC)) && !defined(BOOST_GNU_STDLIB)
#include <cstdlib>
#endif 
```

but I'm not entirely sure that is the best way to detect libc++.

Apparently, there seem to be no `BOOST_GNU_STDLIB` counterpart for libc++ like `BOOST_LIBCXX_STDLIB`. I have been asked about it here https://cpplang.slack.com/archives/C27KZLB0X/p1571857778008400

Reply:
@simmplecoder Let's scratch that idea. In order to detect libc++ we need to include a header from it, so you might as well just include `cstdlib` header.